### PR TITLE
ui: aria-menu modifier

### PR DESCRIPTION
**Pull request by johncowen**
New `{{aria-menu}}` modifier, slightly based on our old, and slightly crusty pre-octane [`<AriaMenu>` component](https://github.com/hashicorp/consul/blob/f020cedab29467ed630ac173a0accd3ea5747643/ui/packages/consul-ui/app/components/aria-menu/index.js), which is now deprecated and soon to be removed. The code here has not been completely re-thought, so at some point I'll probably come back and spruce it up a little more. For now its gets us as far as we need to go.

[Rendered Docs](https://consul-ui-staging-1x3l3u5ar-hashicorp.vercel.app/ui/docs/modifiers/aria-menu)

### CODE REVIEW

**Review by randallmorey — CHANGES_REQUESTED** [review]
Can you throw a link into the docs showing the reference used for keyboard navigation?

**Comment by johncowen** [comment]
The keyboard interaction was wholly modelled on GitHubs menubar menus. In the top right of the black bar at the top here.

**Comment by johncowen** [comment]
K added a note in the docs here.

**Review by randallmorey — APPROVED** [review]

**Comment by johncowen** [comment]
Merged the MenuComponent one down here, I'll just wait for tests to pass here in case and then merge down, probably tomorrow now, thanks for the reviews!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/578260.

**Pull request merged**
